packages/network-controller/tests/NetworkController.test.ts

```diff
@@ -5297,764 +5333,885 @@ describe('NetworkController', () => {
   });
 
   describe('rollbackToPreviousProvider', () => {
-    for (const {
-      networkType,
-      chainId,
-      blockExplorerUrl,
-      ticker,
-      nickname,
-      networkVersion,
-    } of INFURA_NETWORKS) {
+    for (const { networkType } of INFURA_NETWORKS) {
       describe(`if the previous provider configuration had a type of "${networkType}"`, () => {
         it('overwrites the the current provider configuration with the previous provider configuration', async () => {
-          const rpcUrlOrTarget = 'https://mock-rpc-url-1';
-          const customNetworkConfiguration = {
-            chainId: '0x1337',
-            nickname: 'test-chain',
-            ticker: 'TEST',
-            rpcPrefs: {
-              blockExplorerUrl: 'test-block-explorer.com',
-            },
-            id: 'testNetworkConfigurationId',
-          };
-
-          const initialProviderConfig = {
-            ...buildProviderConfig({
-              type: networkType,
-              chainId,
-              ticker,
-              rpcPrefs: {
-                blockExplorerUrl,
-              },
-            }),
-          };
-
           await withController(
             {
               state: {
-                providerConfig: initialProviderConfig,
+                providerConfig: {
+                  type: networkType,
+                  // NOTE: This doesn't need to match the logical chain ID
```

Comment:
Nit: Maybe better to use `buildProviderConfig({ ...BUILT_IN_NETWORKS[networkType], type: networkType })` here instead, to keep this more concise and avoid cluttering up the test too much.

All of these extra details makes it harder to tell what condition is being tested, and using an unrealistic initial state (though still possible from an API standpoint) makes the test less similar to production conditions.

Reply:
Good point. Fixed in [f3b02f7](https://github.com/MetaMask/core/pull/1223/commits/f3b02f77a950ed909aab5f835e1bde0af32e1cba).